src/doc/src/reference/resolver.md

```diff
@@ -91,6 +93,31 @@ at the time of this writing) and Package B will use the greatest `0.6` release
 (`0.6.5` for example). This can lead to potential problems, see the
 [Version-incompatibility hazards] section for more details.
 
+The resolver algorithm favors the greatest available versions of dependencies
+and has limited support for backtracking in the solutions it attempts.
```

Comment:
"and has limited support for backtracking in the solutions it attempts." I don't think this is technically true. If there is no valid solution, it can backtrack as much as it needs to to prove that. Potentially trying literally every possible combination of versions. It does not do backtracking to find a "better" solution than the valid solution it currently has.

What are we trying to say with the sentence? And how can we say it more clearly?

Reply:
This is where your familiarity with the actual algorithm is probably needed to be more precise. The intention is to temper expectations set by an earlier sentence:

> When multiple packages specify a dependency for a common package, the resolver attempts to ensure that they use the same version of that common package,

... which would make me expect the example to resolve to a non-duplicated solution. Would it be fair to say that "the resolver does not optimize to reduce crate duplications", "does not have a deduplication pass", or something along those lines?

Reply:
That makes sense. In effect were trying to elaborate on the part of the quote you left out:

> When multiple packages specify a dependency for a common package, the resolver
attempts to ensure that they use the same version of that common package, as
long as they are within a SemVer compatibility range.

Specifically within open requirement cargo treated as a case of "If multiple packages have a common dependency with semver-incompatible versions..." instead of a case of "When multiple packages specify a dependency for a common package, the resolver attempts to ensure that they use the same version of that common package...", and it does not do something fancy like trying to solve for the second and then falling back to the first if there's an error. 🤔 about how best to say that.

Words are hard.

Reply:
I've rephrase this piece fairly significantly.